### PR TITLE
Move profiling docs to the Observability docs

### PR DESCRIPTION
**Pull request by mdbirnstiehl**
# Summary

This PR closes [Issue 2614](https://github.com/elastic/observability-docs/issues/2614).

I have taken the information from the current Universal Profiling GitHub Pages documentation and moved it into the repo with the rest of the Observability Docs.

Still left to do:

- [x] Add beta information to the top of each page.
- [x] Add the information on Adding Symbols for Native Frames to the *Advanced configuration* section. @inge4pres will provide this information.

[link to the docs](https://observability-docs_2785.docs-preview.app.elstc.co/guide/en/observability/master/universal-profiling.html)

### CODE REVIEW

**Comment by inge4pres** [comment]
Hey there 😄 
I pushed a commit in the unofficlal doc repo to fix native frames symbolziation in 8.7.
https://github.com/elastic/universal-profiling-documentation/commit/1c48a6804cca7c2ff27c3ec08b98e5c6c916c47f

File: docs/en/observability/profiling-split-filter-data.asciidoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
+[[profiling-split-filter-data]]
+= Split filter data
+
+`elastic-profiling config` prints a default configuration that allows ingesting data into a {ecloud} deployment.
+The only config setting you can change is `project-id` (default value is `1`).
```
[the remark this answers is not in the document]
Comment (inge4pres) [reply docs/en/observability/profiling-split-filter-data.asciidoc:5]:
```suggestion
The instructions to deploy the host-agent displayed in *Add Data* show a default configuration that allows ingesting data into a {ecloud} deployment.
The only config setting you may want to change is `project-id` (default value is `1`).
```

File: docs/en/observability/profiling-use-a-proxy.asciidoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+You can use the `HTTPS_PROXY` environment variable to configure a proxy used by the host-agent.
+The connection to the backend ({ecloud}) will be tunneled through the proxy (no MITM-TLS) using
+the CONNECT method. Basic authentication is supported. This can be useful in environments with security
+policies or network restrictions that block direct connections to outside hosts. 
+
+The following example shows how to use the `HTTPS_PROXY` environment variable:
+
+```
+export HTTPS_PROXY=http://username:password@proxy:port
+```
```
Comment (inge4pres) [inline docs/en/observability/profiling-use-a-proxy.asciidoc:17]: Also here we should be more verbose that this setting need to be applied in the machines that will run the host-agents, and if the value is applied _after_ host-agent is installed, the host-agent installation may need to be restarted to take effect.

**Review by inge4pres — COMMENTED** [review]
Thanks for pulling this off 🙏🏼 
Great work so far, couple of comments below

File: docs/en/observability/profiling-advanced-configuration.asciidoc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+[[profiling-advanced-configuration]]
+= Advanced configuration
+
+After completing the steps in the <<profiling-get-started, Get started>> documentation, you may need to continue to more advanced configuration for your deployment.
+See the following sections for more information:
+
+* <<profiling-split-filter-data, Split your filter data>>
```
[the remark this answers is not in the document]
Comment (mdbirnstiehl) [reply docs/en/observability/profiling-advanced-configuration.asciidoc:9]: Makes more sense to me

File: docs/en/observability/profiling-troubleshooting.asciidoc [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+NOTE: We recommend only enabling debug logs on a single instance of host-agent rather than an entire deployment to limit the amount of logs produced.
+
+[discrete]
+[[profiling-improve-load-time]]
+== Improve load times
+
+The amount of data loaded for the flamegraph, topN functions, and traces view can lead to latency when using a slow connection (e.g. DSL or mobile).
+
+If you're running Kibana/ES 8.6+, setting the Kibana cluster option `server.compression.brotli.enabled: true` reduces the amount of data transferred and should reduce load time.
```
Comment (danielmitterdorfer) [inline docs/en/observability/profiling-troubleshooting.asciidoc:109]: Given that this guide will only be available starting with 8.7.0 we can remove the statement about the minimum version of 8.6+?

**Review by danielmitterdorfer — COMMENTED** [review]
Thanks! I left a few comments

File: docs/en/observability/profiling-use-a-proxy.asciidoc (hunk 2 above)
Comment (inge4pres) [reply docs/en/observability/profiling-use-a-proxy.asciidoc:19]: Perfect thanks 👍🏼

File: docs/en/observability/universal-profiling.asciidoc [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+One of the key goals of Universal Profiling is to have net positive cost benefit for users: the cost of profiling and
+observing applications should not be higher than the savings produced by the optimizations.
+
+In this spirit, both the host-agent and storage are engineered to use as little resources as possible.
+
+[discrete]
+[[profiling-es-storage]]
+=== {es} storage
+
+The Universal Profiling storage budget is predictable on a per-profiled-core basis. The data we generate at the fixed
```
[the remark this answers is not in the document]
Comment (inge4pres) [reply docs/en/observability/universal-profiling.asciidoc:214]: Maybe missing the in-betweens commas?

```
The data we generate, at the fixed sampling frequency of 20 Hz,
```

File: docs/en/observability/profiling-get-started.asciidoc [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+To set up Universal Profiling on your {ecloud} deployment, you need to:
+- Enable the Universal Profiling app in Kibana 
+- Configure data ingestion
+
+[discrete]
+[[profiling-enable-kibana]]
+=== Enable Universal Profiling in Kibana
+
+You'll receive instructions on enabling Universal Profiling at the email you used in your https://docs.google.com/forms/d/e/1FAIpQLSd-SWVgvhO7Z_jAfaV9_bFGa0dUZPuX0JORzPGS8SDP7G-dVQ/viewform[sign-up form].
```
[the remark this answers is not in the document]
Comment (mdbirnstiehl) [reply docs/en/observability/profiling-get-started.asciidoc:74]: @inge4pres will this still be the process?

File: docs/en/observability/profiling-use-a-proxy.asciidoc [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+Basic authentication is supported. 
+This can be useful in environments with security policies or network restrictions that block direct connections to outside hosts. 
+
+IMPORTANT:  You need to set the `HTTPS_PROXY` environment variable in the machines that will run the host-agents. If you set the environment variable after the host-agent is installed, you may need to reinstall the host-agent.
+
+The following example shows how to use the `HTTPS_PROXY` environment variable:
+
+```
+export HTTPS_PROXY=http://username:password@proxy:port
+```
```
Comment (bmorelli25) [inline docs/en/observability/profiling-use-a-proxy.asciidoc:22]: This should be updated to an asciidoc code block: https://github.com/elastic/docs#code-blocks. Triple backticks do render in asciidoc, but they're pretty ugly with our stylesheet.

Comment (bmorelli25) [reply docs/en/observability/profiling-use-a-proxy.asciidoc:22]: Looks like there are a few others that should be updated as well. BTW, you don't need a million dashes like the docs show. Four is the minimum:

```
[source,bash]
----
export HTTPS_PROXY=http://username:password@proxy:port
----
```

File: docs/en/observability/profiling-troubleshooting.asciidoc [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+<titleabbrev>Troubleshoot</titleabbrev>
+++++
+
+beta::[]
+
+You can use the host-agent logs to find errors.
+
+The following is an example of a _healthy_ host-agent output:
+
+----
```
Comment (bmorelli25) [inline docs/en/observability/profiling-troubleshooting.asciidoc:14]: All of the code blocks should specify a source: This differentiates a block a from a code block and will enable code highlighting (if applicable).

```suggestion
[source,logs]
----

```

File: docs/en/observability/profiling-troubleshooting.asciidoc [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+
+This is because a _taint_ precludes the execution of pods on a node unless the workload has been _tolerated_.
+The Helm chart `tolerations` key in the `values.yaml` sets the toleration of taints using the official {k8s} scheduling API
+format.
+
+The following examples provide a `tolerations` config that you can add to the Helm chart `values.yaml`:
+
+* To deploy the host-agent on all nodes with taint `workload=python:NoExecute`, add the following to the `values.yaml`:
++ 
+----
```
Comment (bmorelli25) [inline docs/en/observability/profiling-troubleshooting.asciidoc:135]: Another good example. This one should be source,yaml

```suggestion
[source,yaml]
----
```

Comment (bmorelli25) [reply docs/en/observability/profiling-troubleshooting.asciidoc:135]: I'm not going to comment on all of them—you get the idea :)

File: docs/en/observability/profiling-get-started.asciidoc [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+| {es} | 64 GB | 2
+| Kibana | 8 GB | 1
+| Integrations Server | 8 GB | 1
+|====
+
+Even if you're profiling a smaller fleet, we recommend configuring at least two zones for Elasticsearch and 4 GB of memory each for the Integrations Server and Kibana.
+
+[discrete]
+[[profiling-set-up-on-cloud]]
+== Set up Universal Profiling on a {ecloud} deployment
```
Comment (bmorelli25) [inline docs/en/observability/profiling-get-started.asciidoc:64]:
```suggestion
== Set up Universal Profiling on an {ecloud} deployment

```

File: docs/en/observability/profiling-tag-data-query.asciidoc [hunk 11]
```diff
@@ -0,0 +1,30 @@
+[[profiling-tag-data-query]]
+= Tag data for querying
+
+beta::[]
+
+The instructions to deploy the host-agent displayed in *Add Data* show a default configuration that allows ingesting data into a {ecloud} deployment.
```
Comment (bmorelli25) [inline docs/en/observability/profiling-tag-data-query.asciidoc:6]:
```suggestion
The instructions to deploy the host-agent displayed in *Add Data* show a default configuration that allows ingesting data into an {ecloud} deployment.

```

**Review by bmorelli25 — COMMENTED** [review]
Looking good! Mostly code block recommendations.

File: docs/en/observability/universal-profiling.asciidoc [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+Your stacktrace can be:
+
+* symbolized, showing the full source code's filename and line number
+* partially symbolized
+* not symbolized
+
+In the following screenshot, you can see that unsymbolized frames _do not_ show a filename and line number, but a hexadecimal number such as `0x80d2f4`
+or `<unsymbolized>`.
+
+Adding symbols for unsymbolized frames is currently a manual operation. See *LINK NEEDED*.
```
Comment (danielmitterdorfer) [inline docs/en/observability/universal-profiling.asciidoc:42]: Seems there is a missing link (`LINK NEEDED`)? This should probably point to `profiling-add-symbols.asciidoc`.

File: docs/en/observability/profiling-troubleshooting.asciidoc [hunk 13]
```diff
@@ -0,0 +1,235 @@
...
+
+You can submit a support request from the https://cloud.elastic.co/support[support request page] in the {ecloud} console.
+
+In the support request, specify if your issue deals with the host-agent or the Kibana app.
+
+[discrete]
+[[profiling-send-feedback]]
+== Send feedback
+
+If troubleshooting and support are not fixing your issues, or you have any other feedback that you want to share about the
```
Comment (danielmitterdorfer) [inline docs/en/observability/profiling-troubleshooting.asciidoc:234]: This is probably a question for @iogbole: Do we want to encourage people to send us an email to `profiling-feedback@elastic.co`? Usually we direct people to [our Discuss forum](https://discuss.elastic.co/) and I noticed there is already a [category for profiling](https://discuss.elastic.co/c/observability/profiling/92). Do we want to use this instead once the beta opens?

File: docs/en/observability/profiling-advanced-configuration.asciidoc [hunk 14]
```diff
@@ -0,0 +1,11 @@
+[[profiling-advanced-configuration]]
+= Advanced configuration
+
+beta::[]
+
+After completing the steps in the <<profiling-get-started, Get started>> documentation, you may need to continue to more advanced configuration for your deployment.
+See the following sections for more information:
+
+* <<profiling-tag-data-query, Tag data for querying>>
```
Comment (danielmitterdorfer) [inline docs/en/observability/profiling-advanced-configuration.asciidoc:9]: Nit: The order of this list and the menu on the left are different. Do we want to align them? IMHO a reasonable order is:

* Tag data for querying
* Add symbols for native frames
* Use a proxy

File: docs/en/observability/profiling-add-symbols.asciidoc [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+The executable passed using the `-e` argument is used to calculate the file hash that associates stack traces with their symbols and stripping the binary later changes that hash.
+
+[discrete]
+[[profiling-symbols-linux]]
+== Linux distribution packages
+
+For Debian, Ubuntu, Fedora, and Arch Linux, the `elastic-profiling` tool supports pushing symbols for an entire package at once. 
+The required debug symbols are automatically retrieved from the distribution's https://wiki.debian.org/Debuginfod[debuginfod] server.
+
+For this to work, make sure that the `debuginfod`` client is installed on your machine.
```
Comment (danielmitterdorfer) [inline docs/en/observability/profiling-add-symbols.asciidoc:110]: Nit: The `debuginfod` contains a double-backtick at the end.

File: docs/en/observability/profiling-add-symbols.asciidoc [hunk 16]
```diff
@@ -0,0 +1,151 @@
...
+In the current beta release of Elastic Universal Profiling, native symbolization is still limited
+in a few important ways:
+
+* No virtual frames for inline functions
+* No symbols for leaf frames (top-most frame in a trace)
+* No automatic insertion of debug symbols for OS packages
+
+We hope to improve these limitations and remove them in later versions.
+
+NOTE: If symbols are not displayed in Kibana after ingesting them, try restarting Kibana in your deployment (Cloud → Deployments → `<Deployment Name>` → Kibana → Force Restart). This is a known issue in 8.5 and will be fixed in later versions.
```
Comment (danielmitterdorfer) [inline docs/en/observability/profiling-add-symbols.asciidoc:151]: This note is about a bug in 8.5.0 and I believe it can be removed. @jbcrail can probably advise whether this is definitely fixed.

File: docs/en/observability/profiling-troubleshooting.asciidoc [hunk 17]
```diff
@@ -0,0 +1,235 @@
...
+Failed to probe tracepoint
+----
+* The host-agent is not able to connect to {ecloud}. In this case, a similar message to the following is logged:
++
+[source,logs]
+----
+Failed to setup gRPC connection (retrying...): context deadline exceeded
+----
++
+Verify the `collection-agent` configuration value is set and is equal to what was printed by `elastic-profiling config`.
```
Comment (danielmitterdorfer) [inline docs/en/observability/profiling-troubleshooting.asciidoc:72]: @inge4pres as we now have a single click setup in Kibana, this is outdated, right? Does it make sense to fix this in a separate PR?

File: docs/en/observability/profiling-troubleshooting.asciidoc [hunk 18]
```diff
@@ -0,0 +1,235 @@
...
+rpc error: code = Unauthenticated desc = authentication failed
+----
+* The host-agent is unable to reach the `collectionagent` service. In this case, a similar message to the following is logged:
++
+[source,logs]
+----
+Failed to report hostinfo (retrying...): rpc error: code = Unimplemented desc = unknown service collectionagent.CollectionAgent"
+----
++
+This typically means that your {ecloud} cluster has not been configured for Universal Profiling. To confingure your {ecloud} cluster, follow the steps in <<profiling-configure-data-ingestion,configure data ingestion>>.
```
Comment (danielmitterdorfer) [inline docs/en/observability/profiling-troubleshooting.asciidoc:86]: Nit: `confingure` -> `configure`

File: docs/en/observability/profiling-troubleshooting.asciidoc [hunk 19]
```diff
@@ -0,0 +1,235 @@
...
+----
++
+Verify the `collection-agent` configuration value is set and is equal to what was printed by `elastic-profiling config`.
+* The secret token is not valid, or it has been changed. In this case, the host-agent shuts down, and logs a similar message to the following:
++
+[source,logs]
+----
+rpc error: code = Unauthenticated desc = authentication failed
+----
+* The host-agent is unable to reach the `collectionagent` service. In this case, a similar message to the following is logged:
```
Comment (danielmitterdorfer) [inline docs/en/observability/profiling-troubleshooting.asciidoc:79]: @inge4pres I don't think we ever explain in our docs what the "collection agent" is. Would it make sense to be a bit more vague here and instead write something along the lines of:

```
The host-agent is unable to send data to your deployment. In this case, a similar message to the following is logged:
```

**Review by danielmitterdorfer — COMMENTED** [review]
A few more notes.

File: docs/en/observability/profiling-get-started.asciidoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (inge4pres) [reply docs/en/observability/profiling-get-started.asciidoc:75]: Yes this part was written by @iogbole in the previous doc repo, he asked me to push it here
https://github.com/elastic/universal-profiling-documentation/commit/f3a8b53dd89b633cd91d86b0fbca10ae2996d787

[the remark this answers is not in the document]
Comment (inge4pres) [reply docs/en/observability/profiling-get-started.asciidoc:75]: I'll let @iogbole confirm that 😄

File: docs/en/observability/profiling-troubleshooting.asciidoc (hunk 17 above)
Comment (inge4pres) [reply docs/en/observability/profiling-troubleshooting.asciidoc:72]: Correct. I think we should amend it in this PR.
The instructions are printed directly from Kibana "Add Data", so we could say

```suggestion
Verify the `collection-agent` configuration value is set and is equal to what was printed in Kibana, when clicking to *Add Data*.
```

File: docs/en/observability/profiling-get-started.asciidoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (iogbole) [reply docs/en/observability/profiling-get-started.asciidoc:75]: That's accurate for 8.7. Yes.

File: docs/en/observability/profiling-troubleshooting.asciidoc (hunk 19 above)
Comment (inge4pres) [reply docs/en/observability/profiling-troubleshooting.asciidoc:79]: I guess we were trying to reference the flag passed into the agent, which is still named `collection-agent` (also referenced above).
Your suggestion sounds good Daniel 👍🏼

Comment (inge4pres) [reply docs/en/observability/profiling-troubleshooting.asciidoc:79]: I guess we were trying to reference the flag passed into the agent, which is still named `collection-agent` (also referenced above).
Your suggestion sounds good Daniel 👍🏼

File: docs/en/observability/profiling-add-symbols.asciidoc [hunk 20]
```diff
@@ -0,0 +1,151 @@
...
+to use it.
+
+[discrete]
+[[profiling-use-elastic-profiling]]
+== How to use the `elastic-profiling` binary
+
+Before using the `elastic-profiling` binary, you need the following information for the deployment where you're running Universal Profiling:
+
+* Credentials (either an API key or username and password) for the `superuser` Elasticsearch role (typically, the `elastic` user).
+* Copy the deployment's Cloud ID from the deployment overview page.
```
Comment (inge4pres) [inline docs/en/observability/profiling-add-symbols.asciidoc:23]: Since we're referencing the image, and we said earlier "you need the following information", I would rephrase so

```suggestion
* The deployment's Cloud ID from the deployment overview page.
```

I know this is my own text 😅 ... apologies for that

File: docs/en/observability/profiling-add-symbols.asciidoc [hunk 21]
```diff
@@ -0,0 +1,151 @@
...
++
+[role="screenshot"]
+image::images/profiling-cloud-id.png[]
+* Copy the deployment's APM Component ID from the deployment overview page.
```
Comment (inge4pres) [inline docs/en/observability/profiling-add-symbols.asciidoc:27]:
```suggestion
* The deployment's APM Component ID from the deployment overview page.
```

File: docs/en/observability/profiling-add-symbols.asciidoc [hunk 22]
```diff
@@ -0,0 +1,151 @@
...
++
+[role="screenshot"]
+image::images/profiling-apm-cluster-id.png[]
```
Comment (inge4pres) [inline docs/en/observability/profiling-add-symbols.asciidoc:30]: I'm realizing now this image is out-of-date with the current UI of Cloud.

Here's an updated version.

Screenshot from 2023-03-23 15-48-39

File: docs/en/observability/profiling-troubleshooting.asciidoc (hunk 13 above)
Comment (iogbole) [reply docs/en/observability/profiling-troubleshooting.asciidoc:234]: Good question, @danielmitterdorfer.  The email still applies in 8.7.  We recently created the profiling category in preparation for public beta.

**Review by inge4pres — COMMENTED** [review]
Found couple of more things to correct in the new native frames section.
Thanks for porting it over at the speed of light!

File: docs/en/observability/profiling-add-symbols.asciidoc (hunk 16 above)
Comment (jbcrail) [reply docs/en/observability/profiling-add-symbols.asciidoc:151]: Unfortunately it is still theoretically possible to encounter the bug, but it's less likely now that the Elasticsearch plugin is enabled by default in 8.7 (there are no caches in the path that uses the Elasticsearch plugin). If the plugin was disabled and `elastic-profiling push-symbols` failed to clear the caches, then symbols may not be displayed within the 5-minute TTL window on those caches.

I added a note in https://github.com/elastic/prodfiler/issues/2994 to remove the reference to the bug in this document after the caches are removed.

**Review by bmorelli25 — APPROVED** [review]
🚀

**Review by inge4pres — APPROVED** [review]
Thanks Mike 👏🏼
